source/mainnet/smart-contracts/tutorials/wCCD/wCCD-interacting.rst

```diff
@@ -0,0 +1,748 @@
+
+or
+
+.. code-block:: console
+
+    $./concordium-client contract invoke IMPLEMENTATION --entrypoint operatorOf --parameter-json operatorOf.json --energy 25000 --grpc-port 10001
+
+.. image:: ./images/wCCD_tutorial_7.png
+    :width: 100 %
+
+The ``tokenMetadata`` function
+==============================
+
+.. dropdown:: Input parameters for the ``tokenMetadata`` function (click here)
+
+    Create a ``tokenMetadata.json`` file and insert the following JSON object.
+
+    .. code-block:: json
+
+        [""]
+
+    .. note::
+
+        This empty string is required because of the CIS-2 token standard.
+        The `tokenId` of the wCCD token is the smallest unit possible (an empty string).
+
+You are ready now to invoke the ``tokenMetadata`` function with one of the following commands.
+
+.. code-block:: console
+
+    $./concordium-client contract invoke PROXY --entrypoint tokenMetadata --schema tokenMetadata_fallback_schema.bin --parameter-json tokenMetadata.json --energy 25000 --grpc-port 10001
+
+or
+
+.. code-block:: console
+
+    $./concordium-client contract invoke IMPLEMENTATION --entrypoint tokenMetadata --parameter-json tokenMetadata.json --energy 25000 --grpc-port 10001
+
+.. image:: ./images/wCCD_tutorial_8.png
+    :width: 100 %
+
+The ``supports`` function
+=========================
+
+.. dropdown:: Input parameters for the ``supports`` function (click here)
+
+    Create a ``supports.json`` file and insert the following example JSON object. It will query if
+    the two token standards (``CIS-0`` and ``CIS-2``) are supported by the wCCD token.
+
+    .. code-block:: json
+
+        ["CIS-0","CIS-2"]
+
+    .. note::
+
+        You can find more information about the `CIS-0 standard <https://proposals.concordium.software/CIS/cis-0.html>`_
+        and the `CIS-2 standard <https://proposals.concordium.software/CIS/cis-2.html>`_.
+
+You are ready now to invoke the ``supports`` function with one of the following commands.
+
+.. code-block:: console
+
+    $./concordium-client contract invoke PROXY --entrypoint supports --schema supports_fallback_schema.bin --parameter-json supports.json --energy 25000 --grpc-port 10001
+
+or
+
+.. code-block:: console
+
+    $./concordium-client contract invoke IMPLEMENTATION --entrypoint supports --parameter-json supports.json --energy 25000 --grpc-port 10001
+
+The below screenshot shows the response of querying if the wCCD
+token contract supports the following standards
+``["12345","CIS-0","CIS-2"]``. Its response is that it does
+not support the standard ``12345`` but it supports the ``CIS-0`` and the ``CIS-1`` standards.
+
+.. image:: ./images/wCCD_tutorial_9.png
+    :width: 100 %
+
+State-mutative functions
+------------------------
+
+The protocol has four state-mutative functions (``wrap``, ``unwrap``,
+``transfer``, and ``updateOperator``) that you can invoke on the ``proxy`` contract.
```

Comment:
I also raised the issue already that `invoke` is used by Concordium in two different settings (that could be confusing for users) and added it to the FAQs (on-boarding guide for Ethereum developers):

`invoke` can be the simulation done by `concordium-client`
or
it could mean to update/call a function on-chain.

I used to use the wording to `call` a function on the smart contract but that got voted out because `calling` is done on Ethereum and not on Concordium. I have not a good alternative idea for another name but we use `invoke` at many places in the tutorials to refer to "calling a function" on-chain. I support that we should find a better term and only use `invoke` in one of the two settings ideally.

Reply:
Yea, I remember that you mentioned it. But now it is becoming even more of a problem 😅 Perhaps `concordium-client contract simulate` is the way to go.

Reply:
But let's keep `invoke` for now. As Ales correctly states, we invoke the entry point in both cases. The difference is whether we save the result.